### DISCUSSION TITLE
VAL Makefile problem?

### DISCUSSION THREAD

**Discussion by ko56** [shop-planner/shop3#63] [Q&A] [answered]
It seems to me that the line

`${LEX} -+ src/Parser/pddl+.lex -o src/Parser/lex.yy.cc; ${YACC} $< -o src/pddl+.cpp`

is wrong.   It doesn't agree with the man page for 'flex' (at least on FreeBSD).   I had to change it to

	${LEX} -+ -o src/Parser/lex.yy.cc src/Parser/pddl+.lex; ${YACC} $< -o src/pddl+.cpp

to be able to run 'make'.

**Comment by rpgoldman** [comment]
Good catch, thanks!  Also, I see that there's *another* bug in that Makefile that lets it think it succeeded in running `flex` when, in fact, it failed.

**Comment by rpgoldman** [comment, marked as answer]
By the way, the reason it appeared to work before was that the C++ file output by `flex` was committed into the repo.  This was done because in the past it was hard to ensure that one had a VAL-friendly version of `flex` and `bison`.

**Reply by ko56** [reply]
Ah, I was wondering why I was able to run it before.  I forgot that I updated git in the meantime.

**Reply by rpgoldman** [reply]
Git probably wouldn't have done it. The problem was probably that you ran `make clean`. That would have squashed the copy of `lex.yy.cc`.

**Reply by rpgoldman** [reply]
Probably if you put it back, everything would work again.

**Reply by rpgoldman** [reply]
I will push a fix for this.

**Reply by ko56** [reply]
I did run 'make clean' a few times, because I tried different compilers. clang 10 works fine, by the way.
